crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -583,27 +595,39 @@ impl<'a, B: BitmapSlice> VsockPacket<'a, B> {
             return Err(Error::DescriptorLengthTooSmall);
         }
 
-        // All RX descriptor chains should have a header and a data descriptor.
-        if !chain_head.has_next() {
-            return Err(Error::DescriptorChainTooShort);
-        }
-
         let header_slice = mem
             .get_slice(chain_head.addr(), PKT_HEADER_SIZE)
             .map_err(Error::InvalidMemoryAccess)?;
-        let data_desc = desc_chain.next().ok_or(Error::DescriptorChainTooShort)?;
 
-        if !data_desc.is_write_only() {
-            return Err(Error::UnexpectedReadOnlyDescriptor);
-        }
+        // Starting from Linux 6.2 the virtio-vsock driver can use a single descriptor for both
+        // header and data.
+        let data_slice = if chain_head.len() as usize > PKT_HEADER_SIZE {
```

Comment:
I'm wondering if, for the time being (until we change to multiple `VolatileSlice`s), we should also check here if there's no next descriptor in the chain. If there is another descriptor in the chain, we keep the same behaviour as before, ignore the bytes after the header size in the first descriptor, and use only the second one. This kind of translates to: what faulty behaviour do we want to allow?

Reply:
It makes sense to me in the sense of keeping the same behavior as before, but in my opinion we should not have ignored the bytes after the header in the first descriptor. In theory those are part of the payload.

So for now I agree to try to change as little as possible and check both the length and the not having a next. (So I can restore all the tests I changed as well).

In the future let's fix that and handle the various descriptors simply as a sequence of bytes.

What do you think?

Reply:
> It makes sense to me in the sense of keeping the same behavior as before, but in my opinion we should not have ignored the bytes after the header in the first descriptor. In theory those are part of the payload.
> 
> So for now I agree to try to change as little as possible and check both the length and the not having a next. (So I can restore all the tests I changed as well).
> 
> In the future let's fix that and handle the various descriptors simply as a sequence of bytes.
> 
> What do you think?

Yup, I agree. I don't think we can achieve that without the multiple VolatileSlices API change, so we should prioritise this refactoring.

Reply:
I will open a separate issue for that, I see your PR will close the one opened by you, which is okay I think.